includes/Plugin.php

```diff
@@ -143,6 +143,13 @@ public function register() {
 		$this->discovery = new Discovery();
 		add_action( 'init', [ $this->discovery, 'init' ] );
 
+		// Register activation flag logic outside of 'init' since it hooks into
+		// plugin activation.
+		$activation_flag = new Activation_Flag();
+		$activation_flag->register();
+
+		$activation_notice = new Activation_Notice( $activation_flag );
+		$activation_notice->register();
```

Comment:
```suggestion
		$this->activation_notice = new Activation_Notice( $activation_flag );
		$this->activation_notice->register();
```

Reply:
Should these be regsitered on init?

Reply:
There's a comment above that explains why they should **not** :)